README.md

```diff
@@ -3,14 +3,615 @@
 Elastic Search engine
 
 ###Install:
+
+##### Cli script based approach
 ```
 sudo php vendor/oat-sa/lib-tao-elasticsearch/bin/activateElasticSearch.php <pathToTaoRoot> <host> <port> <login> <password>
 ```
  - `pathToTaoRoot` it's root path of your tao
- - `host` it's host of your elasticsearch environment. `localhost` by defaut.
+ - `host` it's host of your elasticsearch environment. `localhost` by default.
  - `port` it's port of your elasticsearch environment. `9200` by default.
  - `login` it's login for you elasticsearch environment. Optional property.
  - `password` it's password for you elasticsearch environment. Optional property.
+ 
+##### Seed file based approach
+
+Following section to be included into seed file and describes an engine, connectivity string, service configuration and fallback  
+```
+    "tao": {
+      "search": {
+        "type": "configurableService",
+        "class": "oat\\tao\\elasticsearch\\ElasticSearch",
```

Comment:
Good job! 

Would it be possible to have those indexes loaded from external json file. Or is it too complex? I am just asking, cause this is bigger than the setup.json file and could grow even more.

Reply:
I would do something like:

- class `oat\tao\elasticsearch\ElasticSearch` has a new option `indexFiles`
- Inside the class we read these json files. The config would be something like:

```
    "tao": {
      "search": {
        "type": "configurableService",
        "class": "oat\\tao\\elasticsearch\\ElasticSearch",
        "options": {
          "hosts": [
            {
              "host": "http://localhost",
              "port": 9200
            }
          ],
          "settings": {
            "analysis": {
              "filter": {
                "autocomplete_filter": {
                  "type": "edge_ngram",
                  "min_gram": 1,
                  "max_gram": 100
                }
              },
              "analyzer": {
                "autocomplete": {
                  "type": "custom",
                  "tokenizer": "standard",
                  "filter": [
                    "lowercase",
                    "autocomplete_filter"
                  ]
                }
              }
            }
          },
          "indexFiles": [
               "path/to/my/index1.json",
               "path/to/my/index2.json",
               "path/to/my/indexN.json"
          ],
          "oat\\tao\\model\\search\\strategy\\GenerisSearch": {
            "type": "configurableService",
            "class": "oat\\tao\\model\\search\\strategy\\GenerisSearch"
          }
        }
      }
    }
```

Wdyt?

Reply:
I think this will be require changing an existing code